libs/unity/library/Runtime/Scripts/Media/WorkQueue.cs

```diff
@@ -37,7 +44,14 @@ public void EnsureIsMainAppThread()
         /// <param name="action">The action to execute.</param>
         public void InvokeOnAppThread(Action action)
         {
-            _mainThreadWorkQueue.Enqueue(action);
+            if (_mainAppThread != null && IsMainAppThread)
+            {
+                action();
+            }
+            else
+            {
+                _mainThreadWorkQueue.Enqueue(action);
```

Comment:
Document that calls before `Awake` will enqueue until awoken even if on the correct thread, and not run synchronously?

Reply:
Fixed.